### PR TITLE
Optimise search view

### PR DESCRIPTION
**Pull request by eicher31**
Add "filter/group by" in the search view in sponsorship_correspondence_view.xml

### CODE REVIEW

File: sbc_compassion/models/sponsorship_correspondence.py [hunk 1]
```diff
@@ -104,6 +104,8 @@ class SponsorshipCorrespondence(models.Model):
         store=True,
         readonly=True)
     marked_for_rework = fields.Boolean()
+        readonly=True,
```
Comment (ecino) [inline sbc_compassion/models/sponsorship_correspondence.py]: Oups, something went wrong during rebase... We lost field definition.

**Comment by eicher31** [comment]
I fixed the issue

File: sbc_compassion/views/sponsorship_correspondence_view.xml [hunk 2]
```diff
@@ -83,6 +83,28 @@
...
 
+        <!-- Extends the search on sponsorship correspondence -->
+        <record id="sponsorship_correspondence_view_search" model="ir.ui.view">
+            <field name="name">sponsorship.correspondence.search</field>
+            <field name="model">sponsorship.correspondence</field>
+            <field name="arch" type="xml">
+                <search>
+                    <filter name="direction" string="Supporter To Beneficiary" domain="[('direction','=','Supporter To Beneficiary')]" />
+                    <separator />
+                    <filter name="direction" string="Beneficiary To Supporter" domain="[('direction','=','Beneficiary To Supporter')]" />
```
Comment (ecino) [inline sbc_compassion/views/sponsorship_correspondence_view.xml]: You should not have two filters with same name

File: sbc_compassion/views/sponsorship_correspondence_view.xml [hunk 3]
```diff
@@ -83,6 +83,28 @@
...
+
+                    <group expand="0" string="Group By">
+                        <filter string="Partner" domain="[]" context="{'group_by': 'correspondant_id'}" />
```
Comment (ecino) [inline sbc_compassion/views/sponsorship_correspondence_view.xml]: Can you add group by sponsorship and group by state as well ?

**Comment by eicher31** [comment]
It's done

**Comment by ecino** [comment]
:+1:

**Comment by ecino** [comment]
I know this is not related to your code, but can you fix the following PEP8 issues (strange they were not detected before) :
./onramp_compassion/controllers/json_request.py:32:13: W503 line break before binary operator
./sponsorship_tracking/model/ir_ui_menu.py:25:21: W503 line break before binary operator

Thanks

**Comment by eicher31** [comment]
I haven't issues on this files in my repository

**Comment by ecino** [comment]
If you use latest version of PEP8 rules, you will see them appear:
```
sudo pip install flake8 --upgrade
flake8 --exclude=__init__.py compassion-modules
```

**Pull request merged**